mkosi/qemu.py

```diff
@@ -679,7 +729,27 @@ def finalize_firmware_variables(
             if config.qemu_firmware_variables == Path("microsoft") or not config.qemu_firmware_variables
             else config.qemu_firmware_variables
         )
-        shutil.copy2(vars, Path(ovmf_vars.name))
+        if (config.secure_boot_certificate and
+                (config.qemu_firmware_variables == Path("microsoft") or not config.qemu_firmware_variables)):
```

Comment:
As mentioned in Telegram, I would really prefer we don't conditionalize based on whether there's a secure boot certificate configured or not as that's build time configuration and not runtime configuration. So if you'd like to have this for the microsoft keys as well please gate it behind a new constant for `QemuFirmwareVariables=` named `microsoft+mok`